packages/gtm-enhanced/src/index.ts

```diff
@@ -0,0 +1,211 @@
+  );
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/tag-manager
+ *
+ * @api public
+ */
+
+EnhancedGTM.prototype.initialize = function() {
+  if (process.env.NODE_ENV === "test") {
+    this.ready();
+  } else {
+    push({ "gtm.start": Number(new Date()), event: "gtm.js" });
+
+    if (this.options.environment.length) {
+      this.load("with-env", this.options, this.ready);
+    } else {
+      this.load("no-env", this.options, this.ready);
+    }
+  }
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+EnhancedGTM.prototype.loaded = function() {
+  if (process.env.NODE_ENV === "test") {
+    return !!window["dataLayer"];
+  }
+
+  return !!(
+    window["dataLayer"] && Array.prototype.push !== window["dataLayer"].push
+  );
+};
+
+/**
+ * Page.
+ *
+ * @api public
+ * @param {Page} page
+ */
+EnhancedGTM.prototype.page = function(page: any) {
+  const category = page.category();
+  const name = page.fullName();
+  const opts = this.options;
+
+  // all
+  if (opts.trackAllPages) {
+    this.track(page.track());
+  }
+
+  // categorized
+  if (category && opts.trackCategorizedPages) {
+    this.track(page.track(category));
+  }
+
+  // named
+  if (name && opts.trackNamedPages) {
+    this.track(page.track(name));
+  }
+};
+
+/**
+ * Track.
+ *
+ * https://developers.google.com/tag-manager/devguide#events
+ *
+ * @api public
+ * @param {Track} track
+ */
+EnhancedGTM.prototype.track = function(track: any) {
+  const props = track.properties();
+  props.event = track.event();
+
+  push({ ...enhancedUserInfo(this.analytics, this.options), ...props });
+};
+
+/**
+ * Product Clicked.
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-data
+ *
+ * @api public
+ * @param {Track} track
+ */
+EnhancedGTM.prototype.productClicked = function(track: any) {
+  const userProps = enhancedUserInfo(this.analytics, this.options);
+  const product = enhancedEcommerceTrackProduct(track, this.options);
+
+  push({
+    ...userProps,
+    event: "productClick",
+    ecommerce: {
+      click: {
+        products: [product],
+      },
+    },
+  });
+};
+
+EnhancedGTM.prototype.productAdded = function(track: any) {
+  const userProps = enhancedUserInfo(this.analytics, this.options);
+  const product = enhancedEcommerceTrackProduct(track, this.options);
+
+  push({
+    ...userProps,
+    event: "addToCart",
+    ecommerce: {
+      currencyCode: "USD",
+      add: {
+        products: [product],
+      },
+    },
+  });
+};
+
+/**
+ * Completed checkout step - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#checkout-options
+ *
+ * @api private
+ */
+
+EnhancedGTM.prototype.checkoutStarted = function(track) {
+  const userProps = enhancedUserInfo(this.analytics, this.options);
+
+  push({
+    ...userProps,
+    event: "checkout",
+    ecommerce: {
+      checkout: {
+        actionField: { step: 1 },
```

Comment:
let's use the `step` from `track.properties()`

Reply:
For the checkout started event here, I set to 1 since it will always be the first step. Then for the `analytics.track('Checkout Step Viewed`, {...}) track methods I used the `track.properties().step` to dynamically handle the steps 2-3 👍